src/Raven.Server/SqlMigration/RavenWriter.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Handlers;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using CommandType = Raven.Client.Documents.Commands.Batches.CommandType;
+
+namespace Raven.Server.SqlMigration
+{
+    public delegate void TableWrittenEventHandler(string tableName, double time);
+
+    internal class RavenWriter : IDisposable
+    {
+        public event TableWrittenEventHandler OnTableWritten;
+
+        private readonly DocumentsOperationContext _context;
+        private readonly SqlDatabase _database;
+
+        public RavenWriter(DocumentsOperationContext context, SqlDatabase database)
+        {
+            if (!database.IsValid())
+                throw new InvalidOperationException("Database is not valid");
+
+            _database = database;
+            _context = context;
+        }
+
+        public async Task WriteDatabase()
+        {
+            var sw = new Stopwatch();
+
+            foreach (var table in _database.Tables)
+            {
+                if (table.IsEmbedded)
+                    continue;
+
+                sw.Restart();
+
+                using (var reader = table.GetReader())
+                {
+                    while (reader.Read())
+                    {
+                        var document = _database.Factory.FromReader(reader, table, out var attachments);
+
+                        try
+                        {
+                            await InsertDocument(document, attachments);
+                        }
+                        catch (Exception)
+                        {
+                            if (_database.Factory.Options.SkipUnsopportedTypes)
```

Comment:
This is pretty bad, you don't know what the error is, and you aren't logging anything.
But you are making decisions based on a property that isn't related based on the naming